### DISCUSSION TITLE
PEGylated proteins and issues with from_itp/multiblock

### DISCUSSION THREAD

**Discussion by JJJZISO-zZ** [marrink-lab/polyply_1.0#261] [Q&A] [answered]
Hi Developer,

I recently tried to connect DNA with other blocks. I did some procedures just like the PEGlyted-Protein Tutorial but it gives me "Block PEG seems to represent more than a single residue........Block PEG seems to represent more than a single residue". I don't really understand how to add such "from_itp" label or how to solve this problem. My command is just the same as the tutorial

">polyply gen_itp -f PEG_50.itp OH_Beads.itp link1.ff -seq PEG:1 OHend:1 -o PEG_50_OH.itp -name PEGOH".

Can you help me with that?

**Comment by fgrunewald** [comment, marked as answer]
Hi @JJJZISO-zZ,

The issue here is that polyply as of version 1.2.0 doesn't support reading of itp-files in combination with the '-seq' option anymore. Bottom line the tutorial is outdated. As mentioned in issue #169 we're working on updating the tutorial. 

However, good news is that instead of having to run multiple commands, you now can do it in two commands. Let me explain a little: What you need is a .json residue graph file of your complete target molecule. In case of the Tutorial we want to PEGylate lysozyme. So we need the graph-file for lysozyme-linker-PEG-OH. Secondly, the residues which correspond to an itp-file need to have the label "from_itp". The problem with the old syntax - which was admittedly easier - is that it can lead to ambiguous situations, where it is not clear how to connect the lysozyme and PEG. 

To generate the graph file you can use the gen_seq auxiliary tool within polyply as follows:
```
polyply gen_seq -f molecule_0.itp -from_file protein:molecule_0\
                           -from_string linker:1:1:MEE-1.0 polymer:50:1:PEO-1.0 end:1:1:OHend-1.0\
                           -seq protein linker polymer end\
                           -connects 0:1:0-0 1:2:0-0 2:3:49-0\
                           -o sequence.json -name test\
                           -label 0:"from_itp":"molecule_0-1."
```
In the above shown command, you provide the itp file with the '-f' option. Then you specify using '-from_file' that the block called protein corresponds to molecule_0 found in the itp file. If you have more than 1 itp file you can just give them all using '-f' and then select the molecules you need using the 'from_file' option. 

Next we define the linker using the from_string syntax, which defines linear or branched blocks using a string. In this case we want the linker to be one block called MEE, then 50 residues of PEO, and 1 block called OHend. 

Next we stick together the sequence using '-seq'. Note that the keywords of the blocks (e.g. protein) match the keywords in the 'from_file' and 'from_string' section. 

Subsequently using '-connects' flag we define how each of the blocks connects. For example '0:1:0-0' means that the first block (index 0) connects to the second block (index 1), within those blocks the first residue of protein connects to the first residue of linker. Note all residues start counting at 0. 

Finally, we have the -label option that labels all nodes of block 0 (i.e. the protein) with the label 'from_itp' that has the value molecule_0 (i.e. the molecule name). The -1 indicates that the probability of having this label is 1. 

To make the itp file you then run, the following command with the files as provided in the tutorial.
```
polyply gen_itp -f molecule_0.itp MEE.itp combined_links.ff PEO.martini.3b.itp .OH_end.itp -seqf sequence.json -o lysoPEG.itp -name lysoPEG
```

We're aware that the syntax and complexity of gen_seq is currently pretty difficult. However, we work on implementing it with a graphical user interface that makes it much easier to generate these graphs. 

Meanwhile, if you want to connect DNA to other blocks, feel free to ask on how to do that using this syntax. It is easier to help with a more complete picture of what you want to do. Finally, please also let us know if you have any trouble running the above command with the tutorial files as provided.

**Comment by fgrunewald** [comment]
I updated the command. See issue #212. @JJJZISO-zZ  would be cool to hear if it works out in your case.
